beps/0001-notifications-system/README.md

```diff
@@ -160,89 +168,202 @@ The following backend service interfaces are added as part of this proposal.
 ```ts
 // TODO - We may want to add an additional wrapping here with interfaces for Notification and NotificationParameters
 
-interface SendNotificationRequest {
-  receivers: SignalReceivers;
+export type NotificationRecipients =
+  | {
+      type: 'entity';
+      entityRef: string | string[]; // typically a user or group entity, otherwise `spec.owner` of the entity/entities
+    }
+  | {
+      type: 'broadcast'; // all users
+    };
 
-  /* TODO - please contribute :) */
-}
+export type NotificationType = 'undone' | 'done' | 'saved';
 
-interface SendNotificationResponse {
+export type Notification = {
   id: string;
-  receivers: SignalReceivers;
+  userRef: string;
+  title: string;
+  description: string;
+  origin: string;
+  link: string;
+  created: Date;
+  saved: boolean;
+  read?: Date;
+  topic?: string;
+  updated?: Date;
```

Comment:
Can you please describe use-case where do we need this for?

Reply:
When using the `scope` parameter and re-using existing notification

Reply:
Is it for the CI-like example mentioned in another comment?
That would be up to presentation to decide, i.e. by ordering the same-scoped messages by creation date.

I vote for not increasing complexity unless necessary.

Reply:
Well one real use case for myself would be in the Q&A plugin when your question gets an answer, the scope would the `question_${id}`. This way it would just update the existing notification instead having multiple rows for all answers for the question. Plus it's not that complex and already implemented in the alternative implementation PR.

Reply:
Another use-case @freben and I discussed yesterday would be updates for an ongoing scaffolder task run. Being able to present the current step along with a timer, e.g. "publishing - 17s ago"

Reply:
I don't think we should use notifications to model database for any application.
We can think of many "such simple" additions. It's not correct.

Moreover, the notification should stay immutable except its metadata like "read", "saved".

Reply:
The scaffolder use case makes sense to me too; the scope would be the task id in this case and would be updated every time scaffolder progresses. I think immutability is not a must for the notifications.